elytron/src/main/java/org/wildfly/extension/elytron/CredentialStoreService.java

```diff
@@ -114,6 +114,9 @@ public void start(StartContext startContext) throws StartException {
             credentialStoreAttributes.put(CS_LOCATION_ATTRIBUTE, loc.toAbsolutePath().toString());
             credentialStore = getCredentialStoreInstance();
             credentialStore.initialize(credentialStoreAttributes, resolveCredentialStoreProtectionParameter(), otherProviders.getOptionalValue());
+            if ( credentialStoreAttributes.get(ElytronDescriptionConstants.CREATE).equals("true") && !loc.toFile().exists() ){
+                credentialStore.flush();
```

Comment:
I hadn't merged it yet but I did see it. :)

https://github.com/wildfly/wildfly-core/pull/2308/files